cloudinit/sources/DataSourceAzure.py

```diff
@@ -651,6 +651,9 @@ def get_public_ssh_keys(self):
         LOG.debug('Retrieving public SSH keys')
         ssh_keys = []
         try:
+            raise KeyError(
+                "Not using public SSH keys from IMDS"
```

Comment:
This message isn't used in the exception handler, so neither the log nor the diagnostic event will indicate the reason for the failure (i.e. it will just log 'Unable to get keys from IMDS, falling back to OVF'). Is the message just here as documentation when reading the code, or was the expectation that it would be exposed outside cloud-init somewhere?

Reply:
Exactly. I put the message in there as self-documenting code in lieu of an actual comment. No expectations of this string being exposed anywhere outside of source code.